### PR TITLE
Getting Started With Solidity and Binance Smart Chain

### LINKED ISSUE #5525: Getting Started With Solidity and Binance Smart Chain
*NOTE*: All **1st-time contributors** should know Topic approval and PR submission does NOT guarantee your Topic/Article will be published. Our team of Peer Reviewers and Content Moderators will review all PRs that come in to make sure they adhere to the standard of quality we expect from the EngEd community. 

We expect all community memebers to go through provided examples, resources, and previously published material before submitting content. As a rule of thumb, please only submit articles (pull requests) that are complete, formatted correctly, and include a fully polished article (ready to be published and error free). 

All revisions and edits should be *completed on your own forked repo (as to not take up room in the queue)*. **Any PR that is submitted incomplete (meaning not ready to be published as is) will be closed.** A PR being closed does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.

### Topic Suggestion
[Language] Getting Started With Solidity and Binance Smart Chain
#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published (subject to vary depending on _volume and/or backlog of articles_).

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

### Writing sample(s): 
Include any links or writing samples - to help our team better gauge your writing quality.

#### Proposed title of article
Getting Started With Solidity and Binance Smart Chain

#### Proposed article introduction

Binance Smart Chain uses the blockchain network protocol to run smart contract-based applications. This platform enables developers to build decentralized applications (DApps) and help users manage their digital assets cross-chain with low latency and large capacity.
Smart contracts use internal code to make decisions about the agreements. Once such popular platform is Binance smart chain. We will use Remix editor, and Solidity to develop an application. This tutorial will explain the basics of tokens, blockchains, smart contracts, and solidity programming language.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
1. Understanding tokens and cryptocurrencies
2. Setting up Solidity and Remix
3. Develop smart contract on Binance Smart Chain

#### Article quality
This is a unique article a unique content that covers blockchain development in a beginner-friendly way. As students, we are constantly learning about this technology.

#### References
N/A

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by EssyWanjira**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5525

### CODE REVIEW

**Comment by EssyWanjira** [comment]
> wave Welcome and thank you very much for contributing your 1st article to the EngEd Program. We always recommend new contributors to go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md), where we have laid out valuable resources meant to help improve our community members' overall technical writing abilities. We are looking forward to going over your amazing content rocket

happy to share what I'm learning :laughing: see you for a review :rocket:

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by EssyWanjira** [comment]
Hello @Bayler, kindly note that Solidity lacks syntax highlighting for markdown. Therefore, I found a stack overflow issue that recommended JS. Thanks for the progress :rocket:

**Comment by Bayler** [comment]
> Hello @Bayler, kindly note that Solidity lacks syntax highlighting for markdown. Therefore, I found a stack overflow issue that recommended JS. Thanks for the progress rocket

Well noted.

File: content/articles/binance-smart-chain-tutorial/index.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Goal
+Tokens are the building blocks of any smart contract-based blockchain. We rely on them to build exchange platforms, Dapps, and Defi platforms. This tutorial provides a context of how we can get started with developing a binance smart chain token using Ethereum compatible tools such as Remix and Metamask. In the process, we will learn the basics of smart contracts, solidity programming, tokens supply, management, and allocation.
+
+### Pre-requisites
+- Basics of [Blockchain]() technology.
+- An IDE for smart contracts. I recommend the in-browser [Remix IDE]().
+- A Metamask wallet extension. [Install here]() on your Chrome browser.
```
Comment (Bayler) [inline content/articles/binance-smart-chain-tutorial/index.md:42]: Please make sure all your links are working. Also, avoid using the word "I" in your tutorial. The focus should always be on the reader. Use, "we" instead.

File: content/articles/binance-smart-chain-tutorial/index.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+In blockchains that support smart contracts, developers need to create crypto tokens that are based on a set of standards. This includes building DApps, exchanges, wallets, etc. While Ethereum's popular standard is the ERC-20 token, BSC uses the BEP-20 standard and in many ways, it is similar to the ERC-20 in Ethereum. Examples of popular BEP-20 tokens include:
+
+- BUSD Token: BUSD stands for [Binance USD](https://paxos.com/busd/). It is a stable coin created by [Paxos](https://paxos.com/), a tether to USD.  
+- CAKE Token: This is a BEP-20 token that controls the [PancakeSwap](https://pancakeswap.finance/) platform.
+- BUX Token was created as a utility token for the [BUX exchange](https://bux-c.com/) ecosystem.
+- SAFEMOON Token: a Defi token for [safemoon](https://safemoon.net/) on the Binance Smart Chain.
+
+### Defining Contracts and Constructor
+
+By convention, each solidity file contains one smart contract. A contract is Solidity is similar to classes in OOP languages in that it wraps a collection of our variables, functions, events, state of a blockchain, and modifiers. Start by defining a contract named `Token` with the following state variables.
```
Comment (Bayler) [inline content/articles/binance-smart-chain-tutorial/index.md:77]:
```suggestion
By convention, each solidity file contains one smart contract. A contract in Solidity is similar to classes in OOP languages in that it wraps a collection of our variables, functions, events, state of a blockchain, and modifiers. Start by defining a contract named `Token` with the following state variables:
```

File: content/articles/binance-smart-chain-tutorial/index.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+### Tokens in Binance Smart Chain
+In blockchains that support smart contracts, developers need to create crypto tokens that are based on a set of standards. This includes building DApps, exchanges, wallets, etc. While Ethereum's popular standard is the ERC-20 token, BSC uses the BEP-20 standard and in many ways, it is similar to the ERC-20 in Ethereum. Examples of popular BEP-20 tokens include:
+
+- BUSD Token: BUSD stands for [Binance USD](https://paxos.com/busd/). It is a stable coin created by [Paxos](https://paxos.com/), a tether to USD.  
+- CAKE Token: This is a BEP-20 token that controls the [PancakeSwap](https://pancakeswap.finance/) platform.
+- BUX Token was created as a utility token for the [BUX exchange](https://bux-c.com/) ecosystem.
+- SAFEMOON Token: a Defi token for [safemoon](https://safemoon.net/) on the Binance Smart Chain.
+
+### Defining Contracts and Constructor
```
Comment (Bayler) [inline content/articles/binance-smart-chain-tutorial/index.md:75]: You have defined a contract but not a constructor

File: content/articles/binance-smart-chain-tutorial/index.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+```js
+    // event transfer
+    event Transfer(address indexed from, address indexed to, uint value);
+    // event emitted during an approval
+    event Approval(address indexed owner, address indexed spender, uint value);
+```
+
+### Create a smart contract constructor
+Next, add a constructor that runs when the contract is executed. Unlike the object-oriented where the constructor gets initialized every time, Solidity invokes only once when the contract is deployed to init the smart contract state.
```
Comment (Bayler) [inline content/articles/binance-smart-chain-tutorial/index.md:129]:
```suggestion
Next, add a constructor that runs when the contract is executed. Unlike in object-oriented programming where the constructor gets initialized every time, Solidity invokes it only once when the contract is deployed to `init` the smart contract state. 
```

File: content/articles/binance-smart-chain-tutorial/index.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+function approve(address spender, uint value) public returns(bool){
+    allowance[msg.sender][spender] = value; // spender can spend *value* amount belonging to sender 
+    emit Approval(msg.sender, spender, value); // emit approval event to allow spending
+    return true;
+}      
+```  
+
+To transfer tokens from an account, we need to call `Approve` on the address of the `spender` within the smart contract itself.
+
+- The line `allowance[msg.sender][spender] = value;`:
```
Comment (Bayler) [inline content/articles/binance-smart-chain-tutorial/index.md:193]: ??

File: content/articles/binance-smart-chain-tutorial/index.md [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+- Finally, we update our mappings and emit the `Transfer` event to complete the transaction.
+
+### A final demo in MetaMask
+
+Using a metamask plugin, we can inject the web3.js namespace within our browser. If we configure and deploy our smart contract in a sandbox test environment, the final app should look like this:
+
+The executable functions that we deploy to the smart contract to run transactions.
+
+![executable functions](/engineering-education/binance-smart-chain-tutorial/execFuncs.png).
```
Comment (Bayler) [inline content/articles/binance-smart-chain-tutorial/index.md:228]: Please make sure you name your files in small letters, i.e., execfuncs.png or exec-funcs.png

**Comment by EssyWanjira** [comment]
working on the changes.

File: content/articles/binance-smart-chain-tutorial/index.md (hunk 3 above)
Comment (EssyWanjira) [reply content/articles/binance-smart-chain-tutorial/index.md:75]: should be Contract alone since smart contract constructor is below :hand:

**Comment by Bayler** [comment]
@EssyWanjira Welcome to our Eng-Ed community. I hope you've taken note of all the commits we've made in this PR. They'll help you better your next article. This PR will now proceed for final review.

**Comment by hectorkambow** [comment]
jumping on this this one

**Pull request merged**